torch/lib/THD/base/Cuda.cpp

```diff
@@ -0,0 +1,23 @@
+#include "Cuda.hpp"
+
+THCState** _THDCudaState;
```

Comment:
It just makes it possible to lazily initialize THC, even after THD. Instead of giving a `THCState*` (which must be valid) to distributed, you can give it a pointer to the place where `THCState*` will be initialized. You're guaranteed that it will be properly set by the time you get a first CUDA tensor in THD.